src/inmanta/data/__init__.py

```diff
@@ -2233,6 +2238,8 @@ class Environment(BaseDocument):
     :param icon: An icon for the environment
     """
 
+    __instance_cache: Dict[uuid.UUID, Dict[bool, "Environment"]] = {}
```

Comment:
Add a comment on why we cache this? I think it's not trivial: one might think we want to avoid db calls but that part should actually be pretty cheap. The part we want to avoid it the acquisition of a new connection, right?